polaris-react/src/components/Button/utils.tsx

```diff
@@ -26,12 +26,22 @@ export function buttonsFrom(
 }
 
 export function buttonFrom(
-  {content, onAction, ...action}: ComplexAction,
+  {content, onAction, plain, destructive, ...action}: ComplexAction,
   overrides?: Partial<ButtonProps>,
   key?: any,
 ) {
+  const variant = !overrides?.variant && plain ? 'plain' : overrides?.variant;
+  const tone = !overrides?.tone && destructive ? 'critical' : overrides?.tone;
```

Comment:
@kyledurand ^

Reply:
Does this still cause TS errors though?

Reply:
No, because the Button gets the correct props set. The `buttonFrom` API doesn't need to change, the bug where variant and tone weren't when the action was configured with properties that should map variant and tone is all that needed to be fixed. So any `Action`ish typed objects implemented in Polaris and Web can remain as they're typed and the buttons will be correct with this patch.

Reply:
Makes sense! We'll still need to, at some point, allow these actions to set tone and variant I think but this is a solid fix for now. That global types file is 🤯